de.dlr.sc.virsat.model.extension.requirements.ui/src/de/dlr/sc/virsat/model/extension/requirements/ui/celleditor/RequirementsAttributeValuePerColumnEditingSupport.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+/**
+ * 
+ */
+package de.dlr.sc.virsat.model.extension.requirements.ui.celleditor;
+
+import org.eclipse.emf.edit.domain.EditingDomain;
+import org.eclipse.jface.viewers.ColumnViewer;
+
+import de.dlr.sc.virsat.model.dvlm.categories.CategoryAssignment;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.APropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.propertyinstances.ComposedPropertyInstance;
+import de.dlr.sc.virsat.model.dvlm.categories.util.CategoryAssignmentHelper;
+import de.dlr.sc.virsat.model.extension.requirements.model.AttributeValue;
+import de.dlr.sc.virsat.model.extension.requirements.model.Requirement;
+import de.dlr.sc.virsat.model.extension.requirements.model.RequirementAttribute;
+import de.dlr.sc.virsat.project.editingDomain.VirSatTransactionalEditingDomain;
+
+/**
+ * Editing support for table with columns for the different attributes
+ *
+ */
+public class RequirementsAttributeValuePerColumnEditingSupport extends AbstractAttributeValueEditingSupport {
+
+	protected final int attributeIndex;
+
+	protected final VirSatTransactionalEditingDomain domain;
+
+	/**
+	 * @param editingDomain
+	 *            the editing domain
+	 * @param viewer
+	 *            the column viewer
+	 * @param attIndex
+	 *            the index of the attribute definition this class should support
+	 */
+	public RequirementsAttributeValuePerColumnEditingSupport(EditingDomain editingDomain, ColumnViewer viewer, int attIndex) {
+		super(editingDomain, viewer, getAttributeValueProperty((VirSatTransactionalEditingDomain) editingDomain));

```

Comment:
Not too much of a fan of this weird constructor setup.
Maybe in the constructor instead of `attIndex` just directl pass the attribute value property?
I think that would make this easier.

Reply:
That does not work because depending on the requirement type of a requirement, there are different attributes with the given index...